kong/client.go

```diff
@@ -164,6 +165,34 @@ func NewClient(baseURL *string, client *http.Client) (*Client, error) {
 	return kong, nil
 }
 
+// WithWorkspace return the client with a defined workspace
+func (c *Client) WithWorkspace(workspace string) *Client {
+	c.SetWorkspace(workspace)
+	return c
+}
+
+// SetWorkspace defines a workspace for the client
+func (c *Client) SetWorkspace(workspace string) {
+	c.workspace = workspace
+}
+
+//ClearWorkspace clear the workspace
+func (c *Client) ClearWorkspace() {
+	c.workspace = ""
+}
+
+// baseURL build the base URL from the rootURL and the workspace
+func (c *Client) baseURL() string {
+	if c.hasWorkspace() {
+		return c.rootURL + "/" + c.workspace
+	}
+	return c.rootURL
+}
+
+func (c *Client) hasWorkspace() bool {
+	return len(c.workspace) > 0
+}
+
```

Comment:
```suggestion
```

This function poses a data race danger because it tempts the user to test `hasWorkspace` and then `GetWorkspace` without synchronization in between, and the data may change between these two calls.